pajbot/modules/wide_emote_limit.py

```diff
@@ -0,0 +1,131 @@
+
+log = logging.getLogger(__name__)
+
+
+class WideEmoteLimitModule(BaseModule):
+    ID = __name__.split(".")[-1]
+    NAME = "Wide Emote Limit"
+    DESCRIPTION = "Punishes users that post too many wide 7TV or FFZ emotes."
+    PAGE_DESCRIPTION = "Users who post too many wide 7TV or FFZ emotes will be punished by this module. Twitch & BTTV don't provide emote size data, so any wide emotes on provided by them will not be accounted for"
+    CATEGORY = "Moderation"
+    SETTINGS = [
+        ModuleSetting(
+            key="max_wide_emotes",
+            label="Maximum number of wide emotes that can be posted",
+            type="number",
+            required=True,
+            placeholder="",
+            default=15,
+            constraints={"min_value": 1, "max_value": 167},
+        ),
+        ModuleSetting(
+            key="emote_max_width",
+            label="Maximum width of emotes in pixels. Emotes exceeding this width will be counted as wide. For example: Setting this value to 130 means any emote with 131 pixels width or more will be counted as wide",
+            type="number",
+            required=True,
+            placeholder="",
+            default=128,
+            constraints={"min_value": 112, "max_value": 1000},
+        ),
+        ModuleSetting(
+            key="bypass_level",
+            label="Level to bypass module",
+            type="number",
+            required=True,
+            placeholder="",
+            default=420,
+            constraints={"min_value": 100, "max_value": 1000},
+        ),
+        ModuleSetting(
+            key="moderation_action",
+            label="Moderation action to apply",
+            type="options",
+            required=True,
+            default="Timeout",
+            options=["Delete", "Timeout"],
+        ),
+        ModuleSetting(
+            key="timeout_online",
+            label="Online timeout duration (if moderation action is timeout)",
+            type="number",
+            required=True,
+            placeholder="",
+            default=60,
+            constraints={"min_value": 1, "max_value": 1209600},
+        ),
+        ModuleSetting(
+            key="timeout_offline",
+            label="Offline Timeout duration (if moderation action is timeout)",
+            type="number",
+            required=True,
+            placeholder="",
+            default=60,
+            constraints={"min_value": 1, "max_value": 1209600},
+        ),
+        ModuleSetting(
+            key="allow_subs_to_bypass",
+            label="Allow subscribers to bypass",
+            type="boolean",
+            required=True,
+            default=False,
+        ),
+        ModuleSetting(
+            key="enable_in_online_chat", label="Enabled in online chat", type="boolean", required=True, default=True
+        ),
+        ModuleSetting(
+            key="enable_in_offline_chat", label="Enabled in offline chat", type="boolean", required=True, default=True
+        ),
```

Comment:
This isn't needed anymore, since setting `timeout_offline` or `timeout_online` should turn them off respectively.